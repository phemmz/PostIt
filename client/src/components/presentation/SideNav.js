import React, { Component } from 'react';

class SideNav extends Component {
	render() {
		return (
      <ul id="slide-out" className="side-nav groupdetails-nav" >
        <li>
          <div className="user-view">
            <div className="background">
              <img className="groupdetails-img" src="https://images.unsplash.com/photo-1493219686142-5a8641badc78?dpr=2&auto=format&fit=crop&w=1500&h=1000&q=80&cs=tinysrgb&crop="/>
            </div>
            <div>
              <a href="#!user"><img className="circle" src=""/></a>
              <a href="#!name"><span className="white-text name">{this.props.currentUser.username}</span></a>
              <a href="#!email"><span className="white-text email right">{this.props.currentUser.email}</span></a>
            </div>
          </div>
        </li>
        <li><a href="#!" className="waves-effect waves-light green nav-anchor"><i className="material-icons">cloud</i>Group Members</a></li>
        <li><div className="divider"></div></li>
        <li><a href="#addUser" className="waves-effect waves-light green nav-anchor modal-trigger"><i className="material-icons">people_outline</i>Add User</a></li>
      </ul>
		)
	}
}
export default SideNav;