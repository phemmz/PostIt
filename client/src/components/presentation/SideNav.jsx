import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router';

const SideNav = ({
  currentUser
}) => {
  return (
    <ul id="slide-out" className="side-nav groupdetails-nav" >
      <li>
        <h5
          className="white-text name user-size current-username"
        >{currentUser.username}</h5>
      </li>
      <li>
        <h5
          className="white-text email user-size current-email"
        >{currentUser.email}</h5>
      </li>
      <li>
        <Link
          to="/members"
          className="waves-effect waves-light green nav-anchor"
        >
          <i
            className="fa fa-users snav"
            aria-hidden="true"
          />Members</Link></li>
      <li><div className="divider" /></li>
      <li>
        <a
          href="#addUser"
          className="waves-effect waves-light green nav-anchor modal-trigger"
        >
          <i
            className="fa fa-user-plus snav"
            aria-hidden="true"
          />Add User</a></li>
    </ul>
  );
};

SideNav.propTypes = {
  currentUser: PropTypes.shape({
    username: PropTypes.string,
    email: PropTypes.string
  }).isRequired
};

export default SideNav;
