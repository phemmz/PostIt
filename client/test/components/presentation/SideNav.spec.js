import React from "react";
import { mount } from "enzyme";
import SideNav from '../../../src/components/presentation/SideNav';
import { user } from '../../__mockData__/dummyData';

describe("SideNav Component", () => {
  let props;
  let mountedSideNav;
  const currentUser = user[0];

  const sideNav = () => {
    if (!mountedSideNav) {
      mountedSideNav = mount(
        <SideNav {...props} />
      );
    }
    return mountedSideNav;
  }

  beforeEach(() => {
    props = {
      currentUser
    };
    mountedSideNav = undefined;
  });

  it("should always render ul", () => {
    const divs = sideNav().find("ul");
    expect(sideNav()).toMatchSnapshot();
    expect(divs.length).toBeGreaterThan(0);
  });

  it("should contain everything else that gets rendered in the ul", () => {
    const divs = sideNav().find("ul");
    const wrappingDiv = divs.first();
    expect(wrappingDiv.children()).toEqual(sideNav().children());
  });

  it("should receive a prop", () => {
    expect(Object.keys(sideNav().props()).length).toBe(1);
  });

  describe("when currentUser prop is defined", () => {
    it("should display the username", () => {
      expect(sideNav().find('.current-username').text()).toBe("femz");
    });
    it("should display the email", () => {
      expect(sideNav().find('.current-email').text()).toBe("femz@gmail.com");
    });
  });
});
