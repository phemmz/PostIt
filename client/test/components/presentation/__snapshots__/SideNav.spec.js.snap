// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNav Component should always render ul 1`] = `
<SideNav
  currentUser={
    Object {
      "email": "femz@gmail.com",
      "id": 1,
      "password": "123456",
      "phoneNo": "08012345678",
      "token": "1212112",
      "username": "femz",
    }
  }
>
  <ul
    className="side-nav groupdetails-nav"
    id="slide-out"
  >
    <li>
      <h5
        className="white-text name user-size current-username"
      >
        femz
      </h5>
    </li>
    <li>
      <h5
        className="white-text email user-size current-email"
      >
        femz@gmail.com
      </h5>
    </li>
    <li>
      <Link
        className="waves-effect waves-light green nav-anchor"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/members"
      >
        <a
          className="waves-effect waves-light green nav-anchor"
          onClick={[Function]}
          style={Object {}}
        >
          <i
            aria-hidden="true"
            className="fa fa-users snav"
          />
          Members
        </a>
      </Link>
    </li>
    <li>
      <div
        className="divider"
      />
    </li>
    <li>
      <a
        className="waves-effect waves-light green nav-anchor modal-trigger"
        href="#addUser"
      >
        <i
          aria-hidden="true"
          className="fa fa-user-plus snav"
        />
        Add User
      </a>
    </li>
  </ul>
</SideNav>
`;
